### PR TITLE
Manage JVM-Heap

### PR DESCRIPTION
**Pull request by melvin-suter**
Should work in theory, will test it soon.
(PR for issue)

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -9,6 +9,11 @@ graylog3_allow_highlighting: true
 graylog3_dashboard_widget_default_cache_time: 60s
 graylog3_install_enterprise_plugin: false
 
+# --- JVM HEAP SIZE ---
+graylog3_jvm_elasticsearch_heap: 1024m
```
Comment (joe-speedboat) [inline defaults/main.yml]: would make sense to use the ansible facts for defaults?
{{ (vars.ansible_memtotal_mb/2)|round|int }}m

File: defaults/main.yml [hunk 2]
```diff
@@ -9,6 +9,11 @@ graylog3_allow_highlighting: true
...
+graylog3_jvm_graylog_heapmax: 1g
```
Comment (joe-speedboat) [inline defaults/main.yml]: 2g sould be fine

File: defaults/main.yml [hunk 3]
```diff
@@ -9,6 +9,11 @@ graylog3_allow_highlighting: true
...
+graylog3_jvm_graylog_heap: 1g
```
Comment (joe-speedboat) [inline defaults/main.yml]: 2g sould be fine

File: tasks/elasticsearch.yml [hunk 4]
```diff
@@ -21,4 +21,21 @@
     name: elasticsearch
     state: started
     enabled: yes
+
+# Addition ; config heap-size
+- name: Configure elasticsearch heap-size
+  lineinfile:
+    path: /etc/sysconfig/elasticsearch
```
Comment (joe-speedboat) [inline tasks/elasticsearch.yml]: this is the wrong place, isn't it?
should be: /etc/elasticsearch/jvm.options
can't find this var in my 6 years old instance under /etc/sysconfig/elasticsearch?
but let's check it tomorrow in lab

File: tasks/graylog.yml [hunk 5]
```diff
@@ -208,4 +208,29 @@
     name: firewalld
     state: restarted
   when: _svc_firewalld.rc == 0 and graylog3_firewalld_setup_rules == true
+
+
+# Addition ; config heap-size
+- name: Configure elasticsearch heap-size
```
Comment (joe-speedboat) [inline tasks/graylog.yml]: its not elasticsearch, its graylog-server here?

File: tasks/graylog.yml [hunk 6]
```diff
@@ -208,4 +208,29 @@
...
+  replace:
+    path: /etc/sysconfig/graylog-server
+    regexp: '-Xms([0-9]+[kmgtKMGT])'
+    replace: "-Xms{{ graylog3_jvm_graylog_heap }} "
+  register: heapsizeline1
+
+- name: Configure elasticsearch heap-size
```
Comment (joe-speedboat) [inline tasks/graylog.yml]: its not elasticsearch, its graylog-server here?

**Comment by melvin-suter** [comment]
I fixed @joe-speedboat 's suggestion except the jvm.options, which I'll test first.

**Comment by melvin-suter** [comment]
Thanks for the beer, but a merge would be even better :p

**Comment by joe-speedboat** [comment]
> Thanks for the beer, but a merge would be even better :p
* shame on me, i'ts not a VI environment :-)

**Comment by joe-speedboat** [comment]
nearly perfect
graylog3_jvm_elasticsearch_heap: "{{ (ansible_memtotal_mb/2-1024)|round|int }}m"

File: defaults/main.yml [hunk 7]
```diff
@@ -9,6 +9,11 @@ graylog3_allow_highlighting: true
 graylog3_dashboard_widget_default_cache_time: 60s
 graylog3_install_enterprise_plugin: false
 
+# --- JVM HEAP SIZE ---
+graylog3_jvm_elasticsearch_heap: {{ (ansible_memtotal_mb/2-1024)|round|int }}m
```
Comment (joe-speedboat) [inline defaults/main.yml]: graylog3_jvm_elasticsearch_heap: "{{ (ansible_memtotal_mb/2-1024)|round|int }}m"

**Pull request merged**